modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldDefaultValue.testcase

```diff
@@ -32,17 +32,181 @@ definition {
 	@description = "LPS-163716 - Verify it's possible to create an object entry with the Picklist default value set via Expression Builder"
 	@priority = 5
 	test CanSetPicklistViaExpressionBuilder {
+		property custom.properties = "feature.flag.LPS-163716=true";
+		property portal.acceptance = "true";
+
+		task ("Given a picklist with several items") {
+			Picklist.addPicklistViaAPI(picklistName = "Custom Picklist");
+
+			for (var itemNumber : list "1,2,3,4,5") {
+				Picklist.addPicklistItemViaAPI(
+					itemKey = "picklistItem${itemNumber}",
+					itemName = "Picklist Item ${itemNumber}",
+					picklistName = "Custom Picklist");
+			}
+		}
 
-		// TODO pending implementation
+		task ("And given a custom object with a text field and a picklist field") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object 181203",
+				objectName = "CustomObject181203",
+				pluralLabelName = "Custom Objects 181203");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Picklist",
+				fieldLabelName = "Field Picklist",
+				fieldName = "fieldPicklist",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "CustomObject181203",
+				picklistName = "Custom Picklist");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Text",
+				fieldLabelName = "Custom Field",
+				fieldName = "customField",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "CustomObject181203");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject181203");
+		}
+
+		task ("And given the default value is set via Expression Builder") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 181203");
+
+			ObjectAdmin.goToFieldsTab();
+
+			ObjectAdmin.goToFieldsDetails(label = "Field Picklist");
+
+			Click(locator1 = "ObjectAdmin#SIDEBAR_ADVANCED_TAB");
+
+			Check.checkToggleSwitch(
+				key_toggleSwitchLabel = "Use Default Value",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+
+			Click(
+				key_optionDefaulValue = "Expression Builder",
+				locator1 = "ObjectAdmin#PICKLIST_INPUT_AS_VALUE_OR_EXPRESSION");
+
+			Type(
+				key_text = "Default Value",
+				locator1 = "TextInput#ANY",
+				value1 = "customField");
+
+			CreateObject.saveObject();
```

Comment:
It was suggested to me not to compress these steps into a macro as they will be repeated a few times.

Reply:
Generally speaking, I prefer to split the steps into the following macros:
- Add macro
- - Click plus icon to add field/entry/etc
- - Use existing generic macros as much as possible instead of creating new ones
- Nagivation macro
- - Navigate to the configuration menu (Example: click kebab icon, click Configuration)
- - Use existing generic macros as much as possible instead of creating new ones
- Configuration macro
- - Not necessary if used only once
- - If used in mutiple tests, use a macro
- Save
- - Use existing generic macros as much as possible instead of creating new ones

This leaves the macros flexible so that we can re-use them in multiple scenarios. What we don't want to do is combine the navigation/configuration/save steps since it limits our options to specific user flows. For example, if we want to create then update, then assert the fields are updated, it'll look like this:
- Add
- Configuration
- Save
- Navigation
- Configuration
- Assertion